src/index.js

```diff
@@ -139,66 +162,117 @@ class MiniCssExtractPlugin {
+              this.getChunks(chunk, chunkGraph)
+            ).filter((module) => module.type === MODULE_TYPE);
+
+            const filenameTemplate =
+              chunk.filenameTemplate || this.options.chunkFilename;
+
+            if (renderedModules.length > 0) {
+              result.push({
+                render: () =>
+                  this.renderContentAsset(
+                    compilation,
+                    chunk,
+                    renderedModules,
+                    compilation.runtimeTemplate.requestShortener
+                  ),
+                filenameTemplate,
+                pathOptions: {
                   chunk,
-                  renderedModules,
-                  compilation.runtimeTemplate.requestShortener
-                ),
-              filenameTemplate: this.options.chunkFilename,
-              pathOptions: {
-                chunk,
-                contentHashType: MODULE_TYPE,
-              },
-              identifier: `${pluginName}.${chunk.id}`,
-              hash: chunk.contentHash[MODULE_TYPE],
-            });
+                  contentHashType: MODULE_TYPE,
+                },
+                identifier: `${pluginName}.${chunk.id}`,
+                hash: chunk.contentHash[MODULE_TYPE],
+              });
+            }
           }
-        }
-      );
+        );
+      } else {
+        compilation.hooks.renderManifest.tap(
+          pluginName,
+          (result, { chunk }) => {
+            const { chunkGraph } = compilation;
+
+            const renderedModules = Array.from(
+              this.getChunks(chunk, chunkGraph)
+            ).filter((module) => module.type === MODULE_TYPE);
+
+            const filenameTemplate =
+              chunk.filenameTemplate || chunk.hasRuntime()
+                ? this.options.filename
+                : this.options.chunkFilename;
+
+            if (renderedModules.length > 0) {
+              result.push({
+                render: () =>
+                  this.renderContentAsset(
+                    compilation,
+                    chunk,
+                    renderedModules,
+                    compilation.runtimeTemplate.requestShortener
+                  ),
+                filenameTemplate,
+                pathOptions: {
+                  chunk,
+                  contentHashType: MODULE_TYPE,
+                },
+                identifier: `${pluginName}.${chunk.id}`,
+                hash: chunk.contentHash[MODULE_TYPE],
+              });
+            }
+          }
+        );
+      }
 
-      compilation.mainTemplate.hooks.hashForChunk.tap(
-        pluginName,
-        (hash, chunk) => {
+      if (
+        typeof webpack.javascript !== 'undefined' &&
+        typeof webpack.javascript.JavascriptModulesPlugin !== 'undefined'
+      ) {
+        webpack.javascript.JavascriptModulesPlugin.getCompilationHooks(
+          compilation
+        ).chunkHash.tap(pluginName, (chunk, hash) => {
```

Comment:
This will be unneeded once the logic uses a RuntimeModule as the content of runtime modules is hashed and added to the chunk hash automatically